patches/api/0441-Expose-native-map-methods-in-PDC.patch

```diff
@@ -0,0 +1,288 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Xander <xander.dekeijzer@student.hu.nl>
+Co-authored-by: TheRealRyGuy <ryan@ryguy.me>
+Date: Tue, 3 Oct 2023 07:28:50 +0200
+Subject: [PATCH] Expose native map methods in PDC
+
+
+diff --git a/src/main/java/org/bukkit/persistence/PersistentDataContainer.java b/src/main/java/org/bukkit/persistence/PersistentDataContainer.java
+index 57609b7793122e135fa0c3b926500849379637b2..b221b97fd540b1f16aea3d6fce72d8a09c7a5885 100644
+--- a/src/main/java/org/bukkit/persistence/PersistentDataContainer.java
++++ b/src/main/java/org/bukkit/persistence/PersistentDataContainer.java
+@@ -9,7 +9,7 @@ import org.jetbrains.annotations.Nullable;
+  * This interface represents a map like object, capable of storing custom tags
+  * in it.
+  */
+-public interface PersistentDataContainer {
++public interface PersistentDataContainer extends Iterable<net.kyori.adventure.key.Key> {
+ 
+     /**
+      * Stores a metadata value on the {@link PersistentDataHolder} instance.
+@@ -153,9 +153,247 @@ public interface PersistentDataContainer {
+     PersistentDataAdapterContext getAdapterContext();
+ 
+     // Paper start
++    /**
++     * Returns an iterator over elements of type {@link net.kyori.adventure.key.Key}.
++     *
++     * @return an Iterator bound to the set of keys from {@link #keySet()}
++     */
++    @Override
++    default @NotNull java.util.Iterator<net.kyori.adventure.key.Key> iterator() {
++        return this.keySet().iterator();
++    }
++
++    /**
++     * Get a set of {@link net.kyori.adventure.key.Key}s present on this {@link PersistentDataContainer}
++     * instance.
++     * <p>
++     * Any changes made to the returned set will not be reflected on the
++     * instance.
++     *
++     * @return the key set
++     */
++    @NotNull
++    Set<net.kyori.adventure.key.Key> keySet();
```

Comment:
I don't think we have to add this either, especially since it doesn't follow `Map#keySet`'s contract. Plus, I don't really see a ton of use for doing this which you can't already do and we don't want to encourage a ton of messing with other plugin's PDC.